### PR TITLE
fix Issue 22035 - [ICE] importC: Segmentation fault parsing invalid case statement

### PR DESCRIPTION
**Pull request by ibuclaw**
The importC fix for #12702.

### CODE REVIEW

File: src/dmd/cparse.d [hunk 1]
```diff
@@ -445,7 +445,7 @@ final class CParser(AST) : Parser!AST
                     // not merged into the current case. This can happen for
                     // case 1: ... break;
                     // debug { case 2: ... }
-                    if (cur.isBreakStatement())
+                    if (cur && cur.isBreakStatement())
```
Comment (MoonlightSentinel) [inline src/dmd/cparse.d:448]: Maybe this check should occur before `statements.push(cur);`?
Just in case further code tries to access the statements

Comment (RazvanN7) [reply src/dmd/cparse.d:448]: Right now the code depends on an array of statements with 1 null element to mean an empty case statement. Putting the check before pushing cur will change that.

Comment (ibuclaw) [reply src/dmd/cparse.d:448]: The rest of the compiler is generally quite resilient to this.

https://github.com/dlang/dmd/blob/ee27b0356052365ab6113eff3f8613228466bab7/src/dmd/statementsem.d#L361-L367

https://github.com/dlang/dmd/blob/ee27b0356052365ab6113eff3f8613228466bab7/src/dmd/s2ir.d#L784

Comment (MoonlightSentinel) [reply src/dmd/cparse.d:448]: I see, thanks for the clarification

**Review by MoonlightSentinel — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**